internal/engine/compiler/compiler_controlflow_test.go

```diff
@@ -679,15 +680,13 @@ func TestCompiler_compileCallIndirect(t *testing.T) {
 
 			// Now that we've generated the code for this function,
 			// add it to the module engine and assign its pointer to the table index.
-			me.functions = append(me.functions, function{
```

Comment:
this appends _sometimes_ change the underlying address stored at the previous iteration of `table[i]` beflow.